### PR TITLE
Fomichev Pavel. Hashing task

### PR DESCRIPTION
**Pull request by PabloDeCortes**

### CODE REVIEW

File: Fomichev/HashingApp/.gitignore [hunk 1]
```diff
@@ -0,0 +1,261 @@
+## Ignore Visual Studio temporary files, build results, and
```
Comment (Andrew414) [inline Fomichev/HashingApp/.gitignore:1]: Niiiice one :). Good template.

File: Fomichev/HashingApp/DataStructs/DataStructs.csproj [hunk 2]
```diff
@@ -0,0 +1,52 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (Andrew414) [inline Fomichev/HashingApp/DataStructs/DataStructs.csproj:1]: I believe this project can have Statistics structure as well

File: Fomichev/HashingApp/HashingApp/MainForm.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+using Implements.Managers;
+using Implements.Services;
+using Implements.Validators;
+using ValidationException = DataStructs.Exceptions.ValidationException;
+
+namespace HashingApp
+{
+    public partial class MainForm : Form
+    {
+        public MainForm() => InitializeComponent();
```
Comment (Andrew414) [inline Fomichev/HashingApp/HashingApp/MainForm.cs:15]: Just curious - is Visual Studio now creating projects like this, with lambda constructors everywhere?
Or did you write it?

File: Fomichev/HashingApp/HashingApp/MainForm.Designer.cs [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            this.ProgramModeComboBox = new System.Windows.Forms.ComboBox();
+            this.HashAlgorithmComboBox = new System.Windows.Forms.ComboBox();
+            this.BlockSizeNumericUpDown = new System.Windows.Forms.NumericUpDown();
+            this.FileDialog = new System.Windows.Forms.OpenFileDialog();
+            this.FileDialogButton = new System.Windows.Forms.Button();
+            this.ProgramModeLabel = new System.Windows.Forms.Label();
+            this.HashAlgorithmLabel = new System.Windows.Forms.Label();
+            this.BlockSizeLabel = new System.Windows.Forms.Label();
+            this.FilePathLabel = new System.Windows.Forms.Label();
+            this.button2 = new System.Windows.Forms.Button();
```
Comment (Andrew414) [inline Fomichev/HashingApp/HashingApp/MainForm.Designer.cs:40]: Main button could have some better name :).

File: Fomichev/HashingApp/HashingApp/MainForm.Designer.cs [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+        /// <summary>
+        /// Required method for Designer support - do not modify
+        /// the contents of this method with the code editor.
+        /// </summary>
+        private void InitializeComponent()
+        {
+            this.ProgramModeComboBox = new System.Windows.Forms.ComboBox();
+            this.HashAlgorithmComboBox = new System.Windows.Forms.ComboBox();
+            this.BlockSizeNumericUpDown = new System.Windows.Forms.NumericUpDown();
```
Comment (Andrew414) [inline Fomichev/HashingApp/HashingApp/MainForm.Designer.cs:33]: I would use ComboBox for block size also, probably with some more readable names, like 
1 KB
2 KB
..
512 KB
1 MB

But your option is also ok

File: Fomichev/HashingApp/HashingApp/Program.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace HashingApp
+{
+    public static class Program
+    {
+        /// <summary>
+        /// The main entry point for the application.
+        /// </summary>
+        [STAThread]
+        public static void Main()
```
Comment (Andrew414) [inline Fomichev/HashingApp/HashingApp/Program.cs:13]: Important notice:
Your program can't be run in pure cmdline mode, without creating windows. Actually, the GUI was expected to be just an addition to the cmdline mode.

**Review by Andrew414 — APPROVED** [review]
I'm really impressed. Very good job!

Score:
3/4 for main task. Two reasons for it: 1) Your program crashes when it can't read or write file (save hashes) and 2) No Command-Line Interface
2/3 for being in time, actually, you are a bit late.
3/3 for additional tasks.

8/10 - the best solution for now.

If you want, you can quickly add command line interface, within one-two days, in this case it will be 9/10. If you take more time for adding cmdline - it will be 4/4 for main task but 1/3 for being in time and 8/10 again.

Thank you,
Andrey Fedchuk

File: Fomichev/HashingApp/HashingApp/MainForm.Designer.cs (hunk 4 above)
Comment (PabloDeCortes) [reply Fomichev/HashingApp/HashingApp/MainForm.Designer.cs:40]: Hah, forgot to add name for this button, sorry)

File: Fomichev/HashingApp/HashingApp/MainForm.cs (hunk 3 above)
Comment (PabloDeCortes) [reply Fomichev/HashingApp/HashingApp/MainForm.cs:15]: I wrote it) Lambda looks amazing

File: Fomichev/HashingApp/HashingApp/Program.cs (hunk 6 above)
Comment (PabloDeCortes) [reply Fomichev/HashingApp/HashingApp/Program.cs:13]: I thought I should replace cmd with GUI. Some misunderstanding happened.

**Pull request merged**